core/src/main/java/org/apache/calcite/avatica/AvaticaStatement.java

```diff
@@ -108,6 +109,7 @@ protected AvaticaStatement(AvaticaConnection connection,
     this.resultSetType = resultSetType;
     this.resultSetConcurrency = resultSetConcurrency;
     this.resultSetHoldability = resultSetHoldability;
+    this.fetchSize = connection.config().fetchSize(); // Default to connection config fetch size.
```

Comment:
let's remove the default value set for the field declaration

Reply:
I'm not clear on what you are suggesting here.  Are you suggesting this be defaulted to 0 and have the real value resolved in getFetchSize()?  I went with the approach I did because a setting made via the connection url is accessed through the connection object.  The connection reference isn't stored within AvaticaStatement so the connection value needs to be propagated down somehow.  You are correct that if the value is explicitly set to 0 either via url param or via Statement.setFetchSize or ResultSet.setFetchSize I have no idea what will happen.  All I can say is that I didn't change that behavior with this change.  It will do whatever is would have done before.